addons/udes_stock_cron/models/stock_picking.py

```diff
@@ -0,0 +1,387 @@
+        """
+        # Assign at the move level disabling refactoring
+        moves = self.mapped("move_lines")
+        moves.with_context(lock_batch_state=True, disable_move_refactor=True)._action_assign()
+
+        # Unreserve any partially reserved lines if not allowed by the picking type
+        moves._unreserve_partial_lines()
+
+        # Refactor after unreserving
+        refactored_moves = moves._action_refactor(stage="assign")
+
+        return refactored_moves.mapped("picking_id")
+
+    def reserve_stock(self):
+        """
+        Reserve stock according to the number of reservable pickings.
+        If this method is called on an empty recordset it will attempt to
+        reserve stock for all eligible picking types.  If the recordset is not
+        empty, it will reserve stock for the picks in the recordset. If the
+        recordset is not empty, it is the callers responsibility to make sure
+        that the pickings belong to at most one batch, otherwise this method
+        cannot respect the priority order of pickings, in this case the
+        behaviour of this method is undefined.
+        In either scenario the picking type flags for reserving complete
+        batches and handling partial batches are respected.
+        The number of reservable pickings is defined on the picking type.
+        0 reservable pickings means this function should not reserve stock
+        -1 reservable picking means all reservable stock should be reserved.
+        """
+
+        Picking = self.env["stock.picking"]
+        PickingType = self.env["stock.picking.type"]
+
+        if self:
+            picking_types = self.mapped("picking_type_id")
+        else:
+            picking_types = PickingType.search(
+                [("active", "=", True), ("u_num_reservable_pickings", "!=", 0)]
+            )
+
+        # We will either reserve up to the reservation limit or until all
+        # available picks have been reserved, depending on the value of
+        # u_num_reservable_pickings.
+        # However we must also take into account the atomic batch reservation
+        # flag (u_reserve_batches) and the handle partial flag
+        # (u_handle_partials).
+
+        for picking_type in picking_types:
+            _logger.info("Reserving stock for picking type %r.", picking_type)
+
+            # We want to reserve batches atomically, that is we will
+            # reserve pickings until all pickings in a batch have been
+            # assigned, even if we exceed the number of reservable pickings.
+            # However, the value of the handle partial flag is false we
+            # should not reserve stock if the batch cannot be completely
+            # reserved.
+            to_reserve = picking_type.u_num_reservable_pickings
+            reserve_all = to_reserve == -1
+            base_domain = [("picking_type_id", "=", picking_type.id), ("state", "=", "confirmed")]
+            limit = 1
```

Comment:
As far this can be a variable i would suggest to define in def reserve_stock with optional variable limit=1 and remove from here maybe.
`def reserve_stock(self, limit=1):`